core/src/main/scala/kafka/server/BrokerMetadataCheckpoint.scala

```diff
@@ -94,11 +103,16 @@ class RawMetaProperties(val props: Properties = new Properties()) {
 }
 
 object MetaProperties {
+  def apply(clusterId: String, nodeId: Int): MetaProperties = {
+    MetaProperties(clusterId, nodeId, MetadataVersions.latest().version())
+  }
+
   def parse(properties: RawMetaProperties): MetaProperties = {
     properties.requireVersion(expectedVersion = 1)
     val clusterId = require(ClusterIdKey, properties.clusterId)
     val nodeId = require(NodeIdKey, properties.nodeId)
-    new MetaProperties(clusterId, nodeId)
+    val metadataVersion = properties.initialMetadataVersion.getOrElse(MetadataVersions.V1.version())
```

Comment:
If `meta.properties` does not have `initial.metadata.version` set (as would be the case with KRaft preview), we treat it as V1